pymeasure/instruments/chroma/chroma62000p.py

```diff
@@ -0,0 +1,156 @@
+from pymeasure.instruments.validators import (
+    truncated_range
+)
+from time import sleep
+import numpy as np
+
+
+
+class Chroma62024p6008(Instrument):
+    """ Represents the Chroma ATE 62000P Programmable DC Source
+    and provides a high-level for interacting with the instrument.
+
+    .. code-block:: python
+
+        chroma = Chroma62024P6008("GPIB::1")
+
+        chroma.apply_current()                # Sets up to source current
+        chroma.source_current_range = 10e-3   # Sets the current range to 10 mA
+        chroma.compliance_voltage = 10        # Sets the compliance voltage to 10 V
+        chroma.source_current = 0             # Sets the source current to 0 mA
+
+        chroma.enable_source()                # Enables the current output
+        chroma.ramp_to_current(5e-3)          # Ramps the current to 5 mA
+
+        chroma.shutdown()                     # Ramps the current to 0 mA and disables output
+
+    """
+
+    def __init__(self, adapter, **kwargs):
+        super(Chroma62024p6008, self).__init__(
+            adapter, "Chroma 62000P Programmable DC Source", **kwargs
+        )
+
+    output_current_limit = Instrument.control(
+        "SOUR:CURR?", "SOUR:CURR %g",
+        """ A floating point property that controls the output current limit setting of the instrument.
+        """,
+        validator=truncated_range,
+        values=[0, 8]
+    )
+
+    get_output_status = Instrument.measurement("CONF:OUTP?", "This function queries the output state of the instrument.")
+
+    output_voltage_level = Instrument.control(
+        "SOUR:VOLT?", "SOUR:VOLT %g",
+        """This function control the output voltage limit of the instrument.
+         """,
+        validator=truncated_range,
+        values=[0, 600]
+    )
+
+    output_ovp_limit = Instrument.control(
+        "SOUR:VOLT:PROT:HIGH?", "SOUR:VOLT:PROT:HIGH %g",
+        """This function control the output current limit of the instrument.
+         """,
+        validator=truncated_range,
+        values=[0, 660]
+    )
+
+    output_ocp_limit = Instrument.control(
+        "SOUR:CURR:PROT:HIGH?", "SOUR:CURR:PROT:HIGH %ga",
+        """This function control the output current of the instrument.
+         """,
+        validator=truncated_range,
+        values=[0, 8.8]
+    )
+
+    output_voltage = Instrument.measurement("FETC:VOLT?", "DC output voltage, in Volts")
+
+    output_current = Instrument.measurement("FETC:CURR?", "DC output voltage, in Amps")
+
+    output_slew_rate = Instrument.control(
+        "SOUR:VOLT:SLEW?", "SOUR:VOLT:SLEW %g",
+        """This function control the output voltage slew rate.
+         """,
+        validator=truncated_range,
+        values=[0.001, 10] # 62012P-600-8: 0.001V/ms - 10V/ms
+    )
+
+    def output_enable(self):
+        self.write("CONF:OUTP ON")
+
+    def output_disable(self):
+        self.write("CONF:OUTP OFF")
+
+    def ramp_to_current(self, current, steps=25, duration=0.5):
```

Comment:
This code is similar to that in the Keithley2400 instrument. https://github.com/ralph-group/pymeasure/blob/f9c4012397e5ea426732d8198f3a8115335de64c/pymeasure/instruments/keithley/keithley2400.py#L417-L432

It would be nice to keep the arguments the same. I think we chose to set the pause to avoid it being set too low. I'm certainly open to using a direct duration, but we should cover that case. It would be nice to refactor this out so all instruments can use a common function, but that can happen in a future PR.